src/Common/DNSResolver.cpp

```diff
@@ -138,16 +140,27 @@ static DNSResolver::IPAddresses resolveIPAddressImpl(const std::string & host)
     return addresses;
 }
 
-static String reverseResolveImpl(const Poco::Net::IPAddress & address)
+static std::vector<String> reverseResolveImpl(const Poco::Net::IPAddress & address)
 {
-    Poco::Net::SocketAddress sock_addr(address, 0);
-
-    /// Resolve by hand, because Poco::Net::DNS::hostByAddress(...) does getaddrinfo(...) after getnameinfo(...)
-    char host[1024];
-    int err = getnameinfo(sock_addr.addr(), sock_addr.length(), host, sizeof(host), nullptr, 0, NI_NAMEREQD);
-    if (err)
-        throw Exception("Cannot getnameinfo(" + address.toString() + "): " + gai_strerror(err), ErrorCodes::DNS_ERROR);
-    return host;
+    boost::asio::io_service io_service;
+    YukiWorkshop::DNSResolver resolver(io_service);
+
+    std::vector<std::string> ptr_records;
+
+    resolver.resolve_a4ptr(boost::asio::ip::address_v4::from_string(address.toString()), [&](int err, auto& hosts, auto&, auto&, uint) {
```

Comment:
(cosmetic: [ClickHouse code style](https://clickhouse.com/docs/en/development/style) says that pointers and references should have a space on either side)